Failed to resolve dependencies for gradle build

### Describe the bug

After generating a quarkus project, I'm unable to build the project using gradle.

I was able to work around the issue by re-ording the maven repositories in the settings.gradle file
```gradle
pluginManagement {
    repositories {
        mavenLocal() // <--- This is the problem
        mavenCentral()
        gradlePluginPortal()
    }
    plugins {
        id "${quarkusPluginId}" version "${quarkusPluginVersion}"
    }
}
rootProject.name='release-workflow-report'
```

Normally, this would be fine, but only if you already have all of the dependencies in you local maven nexus, but if you don't, then you see the error above. To overcome the issue I had to re-order the `repositories` to this
```gradle
pluginManagement {
    repositories {
        mavenCentral()
        gradlePluginPortal()
        mavenLocal() // <--- FTFY
    plugins {
        id "${quarkusPluginId}" version "${quarkusPluginVersion}"
    }
}
rootProject.name='release-workflow-report'
```
After that it was able to build the project and all the dependencies were downloaded. I can understand why you have `mavenLocal()` first, as you don't want to make the round trip to the server each time, but for a first time user this is ridiculous. I was almost ready to give up, but I wanted to try it out.

### Expected behavior

After generating the quarkus project, that I can build the project with gradle

### Actual behavior

I can't build the project because I do not have all of the dependencies locally

### How to Reproduce?

1. Build the project using quarkus CLI
```
quarkus create org.acme:report-generator:1.0.0 --gradle --package-name=report.generator
```
2. Build the project (with no dependencies in your local maven repository)
```
$ ./gradlew build
Downloading https://services.gradle.org/distributions/gradle-7.4.2-bin.zip
...........10%...........20%...........30%...........40%...........50%...........60%...........70%...........80%...........90%...........100%

Welcome to Gradle 7.4.2!

Here are the highlights of this release:
 - Aggregated test and JaCoCo reports
 - Marking additional test source directories as tests in IntelliJ
 - Support for Adoptium JDKs in Java toolchains

For more details see https://docs.gradle.org/7.4.2/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'release-workflow-report'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find guice-4.2.2-no_aop.jar (com.google.inject:guice:4.2.2).
     Searched in the following locations:
         file:/D:/dev/maven-repo/com/google/inject/guice/4.2.2/guice-4.2.2-no_aop.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 3s
```

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-18363 xxxxxx 3.1.7-340.x86_64 2021-03-26 22:17 UTC x86_64 Msys

### Output of `java -version`

openjdk version "11.0.15" 2022-04-19

### GraalVM version (if different from Java)

graalvm-ce-java11-22.1.0

### Quarkus version or git rev

2.9.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.4.2

### Additional information

_No response_

--- a/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/settings.tpl.qute.gradle.kts
+++ b/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/settings.tpl.qute.gradle.kts
@@ -2,9 +2,9 @@ pluginManagement {
     val quarkusPluginVersion: String by settings
     val quarkusPluginId: String by settings
     repositories {
-        mavenLocal()
         mavenCentral()
         gradlePluginPortal()
+        mavenLocal()
 {#if gradle.plugin-repositories}
 {#for rep in gradle.plugin-repositories}
         maven { url = uri("{rep.url}") }

--- a/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle/base/settings.tpl.qute.gradle
+++ b/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle/base/settings.tpl.qute.gradle
@@ -1,8 +1,8 @@
 pluginManagement {
     repositories {
-        mavenLocal()
         mavenCentral()
         gradlePluginPortal()
+        mavenLocal()
 {#if gradle.plugin-repositories}
 {#for rep in gradle.plugin-repositories}
         maven { url "{rep.url}" }

